### DISCUSSION TITLE
DietPi-Software | Allow installing webservers without PHP

### DISCUSSION THREAD

**Discussion by Gill-Bates** [MichaIng/DietPi#5043] [Software Request]
I was wondering that nginx from the Software Catalog comes with PHP 7.4. I would have expected to get only nginx when I select nginx to install.

Wouldn't it be better to split nginx and PHP into two software products? This would have the advantage that only what you need is installed and in addition you could offer more granular PHP-Version 7.4 or PHP 8.

So at the end three separate products:

1. nginx
2. PHP 7.4
3. PHP 8

**Comment by MichaIng** [comment]
That is an old discussion indeed. I personally would split webserver, PHP and database into dedicated selectable software options and even remove the webserver stacks like `LAMP`, but I got overvoted, two times when this topic came up. The ability to install PHP and webserver separately sneaked in with the search function, and the lesser documented CLI, but actually for end users the intention was to have only full stacks selectable and otherwise let `dietpi-software` pick what it needs as dependencies.

Probably I can remove the dependency for PHP from the webservers at least, I'll have a look with one of the future releases, but no ETA as there are other much more urgent tasks.

About PHP versions, if you mean to add [Ondrej's repo](https://deb.sury.org/): We will never ever again offer to install a different PHP version than what is natively shipped with the underlying Debian version. We did that one time to get PHP7.3 up on Stretch, and this caused a whole lot of additional support effort, and does until now, as many users do not know the impact of a 3rd party repository, that they pull in additional PHP instances when installing the meta packages like `php-bcmath`, may have the previous one autoremoved and then break their web applications by this. Furthermore dealing with package incompatibilities (rare but when rolling out to 100,000 users, it sums up, a lot!) between Ondrej's repository and plain Debian was a pain to maintain, required more and more fine grained APT preferences over the time. No one who does not exactly know the impact of such a 3rd party repository which overwrites major Debian packages, should use it, hence we should never again install such via `dietpi-software`.

But what I plan to do is autodetecting an installed PHP version and keep using that one instead of installing an additional one. So one can manually add Ondrej's repo, knowing and living with the consequences, and `dietpi-software` will follow the decision.
